src/signTxUtils.c

```diff
@@ -3,10 +3,39 @@
 #include "uiHelpers.h"
 #include "utils.h"
 #include "signTxUtils.h"
+#include "securityPolicy.h"
+#include "state.h"
+
+static single_account_data_t* singleAccountData = &(instructionState.signTxContext.commonTxData.singleAccountData);
```

Comment:
It seems this is only needed in a single place, let's move it to `violatesAndStoreSingleAccount`.